scipy/stats/_continuous_distns.py

```diff
@@ -9062,6 +9062,191 @@ def _updated_ctor_param(self):
+
+    def _fitstart(self, data):
+        # Default is k=1, but that is not a valid value of the parameter.
+        return super(studentized_range_gen, self)._fitstart(data, args=(2, 1))
+
+    def _munp(self, K, k, df):
+        cython_symbol = '_genstudentized_range_moment'
+        _a, _b = self._get_support()
+        # all three of these are used to create a numpy array so they must
+        # be the same shape.
+        K, k, df = np.atleast_1d(K, k, df)
+
+        def _single_moment(K, k, df):
+            arg = [K, k, df]
+            usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+            llc = LowLevelCallable.from_cython(_stats, cython_symbol, usr_data)
+
+            ranges = [(-np.inf, np.inf), (0, np.inf), (_a, _b)]
+            opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+            return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
+
+        return _single_moment(K, k, df)
+
+    def _pdf(self, x, k, df):
+        cython_symbol = '_genstudentized_range_pdf'
+        _a, _b = self._get_support()
+        x = np.atleast_1d(x)
+
+        def _single_pdf(q, k, df):
+            arg = [q, k, df]
+            usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+            llc = LowLevelCallable.from_cython(_stats, cython_symbol, usr_data)
+
+            ranges = [(-np.inf, np.inf), (_a, _b)]
+            opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+            return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
+
+        ufunc = np.frompyfunc(_single_pdf, 3, 1)
+        return ufunc(x, k, df).astype("float64")
+
+    def _cdf(self, x, k, df):
+        _a, _b = self._get_support()
+        x = np.atleast_1d(x)
+
+        def _single_cdf(q, k, df):
+            # "When the degrees of freedom V are infinite the probability
+            # integral takes [on a] simpler form," and a single asymptotic
+            # integral is evaluated rather than the standard double integral.
+            # (Lund, Lund, page 205)
+            if df < 100000:
+                cython_symbol = '_genstudentized_range_cdf'
+
+                arg = [q, k, df]
+                usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+                llc = LowLevelCallable.from_cython(
+                    _stats, cython_symbol, usr_data)
+
+                ranges = [(-np.inf, np.inf), (_a, _b)]
+                opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+                return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
+
+            else:
+                cython_symbol = '_genstudentized_range_cdf_asymptopic'
+
+                arg = [q, k]
+                usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+                llc = LowLevelCallable.from_cython(
+                    _stats, cython_symbol, usr_data)
+
+                ranges = [(_a, _b)]
+                opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+                return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
```

Comment:
```suggestion
                ranges = [(-np.inf, _b)]
                opts = dict(epsabs=1e-11, epsrel=1e-12)

                return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
```

This is the equation that the asymptotic solver should follow: 
![Screen Shot 2021-03-23 at 8 28 55 PM](https://user-images.githubusercontent.com/44255917/112250331-66650e00-8c16-11eb-9a0f-177e9d089d2e.png)

Integration limits -inf -> inf, but `_get_support()` returns 0 and infinity.

Reply:
Nice catch!